2003-07-28  Alan Modra  <amodra@bigpond.net.au>

	* emulparams/elf64ppc.sh (GOT): Fix for shell brace matching.

	* emulparams/elf64ppc.sh (GOT): Don't merge .toc into .got for ld -r.

--- ld/emulparams/elf64ppc.sh.jj	Wed Jul 23 11:08:12 2003
+++ ld/emulparams/elf64ppc.sh	Mon Jul 28 10:25:00 2003
@@ -18,8 +18,15 @@ OTHER_BSS_SYMBOLS="
   .tocbss	${RELOCATING-0}${RELOCATING+ALIGN(8)} : { *(.tocbss)}"
 OTHER_PLT_RELOC_SECTIONS="
   .rela.tocbss	${RELOCATING-0} : { *(.rela.tocbss) }"
-GOT="
-  .got		${RELOCATING-0}${RELOCATING+ALIGN(8)} : { *(.got .toc) }"
+
+if test x${RELOCATING+set} = xset; then
+  GOT="
+  .got		ALIGN(8) : { *(.got .toc) }"
+else
+  GOT="
+  .got		0 : { *(.got) }
+  .toc		0 : { *(.toc) }"
+fi
 OTHER_GOT_RELOC_SECTIONS="
   .rela.toc	${RELOCATING-0} : { *(.rela.toc) }"
 OTHER_READWRITE_SECTIONS="
